Fixes #34346 - support Openstack URI with paths

ares opened the pull request:
The Keystone URI is processed before we pass it's base form to the fog.
If Keystone is deployed under a sub-uri e.g. /identity like Devstack
does, the path is always ommited. This preserves the path.


<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

ares committed:
Fixes #34346 - support Openstack URI with paths

The Keystone URI is processed before we pass it's base form to the fog.
If Keystone is deployed under a sub-uri e.g. /identity like Devstack
does, the path is always ommited. This preserves the path.

ekohl commented on app/models/compute_resources/foreman/model/openstack.rb:
```diff
@@ -262,7 +262,8 @@ def normalize_vm_attrs(vm_attrs)
     def url_for_fog
       u = URI.parse(url)
-      "#{u.scheme}://#{u.host}:#{u.port}"
+      path = u.path.match(/(.*)\/v\d+.*/)[1]
```
For regexes with `/` in it I commonly use `%r{}`
```suggestion
      path = u.path.match(%r{(.*)/v\d+.*})[1]
```

I'm also wondering if this is guaranteed to only accept URLs with `/vN` in them. Otherwise this will end up raising an exception.

ares commented:
I updated it for the URL without the /v information, at that case path can't really be detected. Added a test covering that and converted to use %r{}. Ready for re-review.

ekohl reviewed (approved):
Tests are still running but looks good to merge to me.

ezr-ondrej reviewed (approved):
Thanks @ares ! :+1:

ezr-ondrej commented:
[test foreman]

lzap reviewed (approved):
Yeah this changed like 5 times already.
